core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -841,6 +860,7 @@ public V apply(final K k, V inCache) {
       store.compute(key, remappingFunction, REPLACE_FALSE);
       if (removed.get()) {
         conditionalRemoveObserver.end(ConditionalRemoveOutcome.SUCCESS);
+        markAllCacheEventsFireable();
```

Comment:
The more I think about the pattern of marking events fireable only on certain path, the more I fear consequences. I am afraid you are _silencing_ potential expiry and / or eviction events.